src/modules/party/party.controller.test.ts

```diff
@@ -64,33 +81,75 @@ describe('PartyController', () => {
-    const { createPartyRequest: newParty } = new CreatePartyGenerator(valueGenerator, dateStringTransformations).generate({ numberToGenerate: 1 });
+    const { apiCreatePartyRequest } = new CreatePartyGenerator(valueGenerator, dateStringTransformations).generate({ numberToGenerate: 1 });
     const res: Response = {
       status: jest.fn().mockReturnThis(),
     } as any;
+    const [newParty] = apiCreatePartyRequest;
+    const { officerRiskDate: ratedDate } = newParty;
+    const { externalRatings } = new GetPartyExternalRatingGenerator(valueGenerator).generate({ numberToGenerate: 1, partyIdentifier });
+    const salesforcePartyWithSovereignAccountType = { account: { type: 'Overseas Government Dept' } };
+    const salesforcePartyWithCorporateAccountType = { account: { type: 'Some corporate account type' } };
+
+    const { SOVEREIGN, CORPORATE } = ENUMS.ASSIGNED_RATING_CODES;
 
     it('creates a party with the service if the PartyAlternateIdentifier does not match existing parties', async () => {
-      await controller.createParty(newParty, res);
+      when(partyServiceCreateParty).calledWith(newParty).mockResolvedValueOnce({ partyIdentifier });
+      await controller.createParty(apiCreatePartyRequest, res);
 
-      expect(partyServiceCreateParty).toHaveBeenCalledWith(newParty[0]);
+      expect(partyServiceCreateParty).toHaveBeenCalledWith(newParty);
     });
 
     it('returns the party identifier if the PartyAlternateIdentifier matches an existing party', async () => {
-      when(partyServiceGetPartyIdentifierBySearchText).calledWith(newParty[0].alternateIdentifier).mockResolvedValueOnce({ partyIdentifier });
+      when(partyServiceGetPartyIdentifierBySearchText).calledWith(newParty.alternateIdentifier).mockResolvedValueOnce({ partyIdentifier });
 
-      const response = await controller.createParty(newParty, res);
+      const response = await controller.createParty(apiCreatePartyRequest, res);
 
       expect(res.status).toHaveBeenCalledWith(200);
       expect(response).toStrictEqual({ partyIdentifier });
     });
 
     it('returns the party identifier if creating the party succeeds', async () => {
-      when(partyServiceGetPartyIdentifierBySearchText).calledWith(newParty[0].alternateIdentifier).mockResolvedValueOnce(undefined);
-      when(partyServiceCreateParty).calledWith(newParty[0]).mockResolvedValueOnce({ partyIdentifier });
+      when(partyServiceGetPartyIdentifierBySearchText).calledWith(newParty.alternateIdentifier).mockResolvedValueOnce(undefined);
+      when(partyServiceCreateParty).calledWith(newParty).mockResolvedValueOnce({ partyIdentifier });
 
-      const response = await controller.createParty(newParty, res);
+      const response = await controller.createParty(apiCreatePartyRequest, res);
 
       expect(response).toStrictEqual({ partyIdentifier });
     });
+
+    it('does not create an external rating for the newly created party if one or more external ratings already exist', async () => {
```

Comment:
Can we have a test for whether or not the external rating is created if the party already existed please?

Reply:
Added tests